io/src/main/scala/sbt/internal/io/EventMonitor.scala

```diff
@@ -0,0 +1,194 @@
+package sbt.internal.io
+
+import java.nio.file.{ ClosedWatchServiceException, Files, Path, WatchKey }
+import java.util.concurrent.ArrayBlockingQueue
+import java.util.concurrent.atomic.{ AtomicBoolean, AtomicInteger }
+
+import sbt.io.WatchService
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+
+private[sbt] sealed trait EventMonitor extends AutoCloseable {
```

Comment:
Could you add the overview of what EventMonitor is as ScalaDoc, plz? (Something similar to what you wrote in commit comment is fine)